### PR TITLE
game of life

### PR DESCRIPTION
**Pull request by dhudomiet**

### CODE REVIEW

File: src/game-of-life/game-of-life.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export default class GameOfLife extends Component {
+    constructor(props) {
+        super(props);
+        this.gameOfLifeApi = new GameOfLifeApi(this.props.length);
+        this.state = {
+            grid: this.gameOfLifeApi.grid,
+            start: false
+        }
+        this.onClick = this.onClick.bind(this);
```
Comment (Surzyn) [inline src/game-of-life/game-of-life.js]: you can use arrow function for that

File: src/game-of-life/game-of-life.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        this.clear = this.clear.bind(this);
+    }
+
+    render() {
+        return (
+            <div>
+                <Grid grid={this.state.grid} onClick={this.onClick} />
+                <div>
+                    <button className='btn btn-sm btn-success' onClick={() => this.setState({grid: this.gameOfLifeApi.calculate()})}>click</button>
```
Comment (Surzyn) [inline src/game-of-life/game-of-life.js]: I think that you should remove arrow function form here and move this to separate function 
Using an arrow function in render creates a new function each time the component renders

File: src/game-of-life/game-of-life-api.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+export default class GameOfLifeApi {
+    constructor(length) {
+        this.length = length;
+        this.grid = Array(length);
+        this.initialize();
+        this.setCellState.bind(this);
```
Comment (Surzyn) [inline src/game-of-life/game-of-life-api.js:6]: don't user bind, change to arrow func

File: src/square.jsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ export default class Square extends Component {
 
     render() {
         return (
-            <button className={this.props.value === 0 ? 'squareWhite' : 'squareBlack'} >
+            <button className={this.props.value === 0 ? 'squareWhite' : 'squareBlack'} onClick={() => this.props.onClick(this.props.row, this.props.col, this.props.value)} >
```
Comment (Surzyn) [inline src/square.jsx]: remove arrow function

File: src/row.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React, {Component} from 'react';
+import Square from "./square";
+
+export default class Row extends Component {
```
Comment (Surzyn) [inline src/row.js:4]: Add shouldComponentUpdate

**Review by Surzyn — CHANGES_REQUESTED** [review]

File: src/game-of-life/game-of-life.js [hunk 6]
```diff
@@ -9,31 +10,43 @@ export default class GameOfLife extends Component {
...
-        this.onClick = this.onClick.bind(this);
-        this.clear = this.clear.bind(this);
+        };
     }
 
     render() {
         return (
             <div>
-                <Grid grid={this.state.grid} onClick={this.onClick} />
+                <ReactInterval timeout={100} enabled={this.state.start} callback={() => this.setState({grid: this.gameOfLifeApi.calculate()})} />
```
Comment (Surzyn) [inline src/game-of-life/game-of-life.js]: remove arrow function

File: src/game-of-life/game-of-life.js [hunk 7]
```diff
@@ -9,31 +10,43 @@ export default class GameOfLife extends Component {
...
                 <div>
-                    <button className='btn btn-sm btn-success' onClick={() => this.setState({grid: this.gameOfLifeApi.calculate()})}>click</button>
+                    <button className='btn btn-sm btn-success' onClick={this.startInterval} >start</button>
+                    <button className='btn btn-sm btn-success' onClick={this.stopInterval} >stop</button>
                     <button className='btn btn-sm btn-success' onClick={this.clear}>clear</button>
                 </div>
+                <Grid grid={this.state.grid} onClick={(i, j, value) => this.setCell(i, j, value)} />
```
Comment (Surzyn) [inline src/game-of-life/game-of-life.js]: remove arrow function

File: src/game-of-life/game.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import Square from "../square";
+
+export default class Game extends Component {
+    constructor(props) {
+        super(props);
+        this.gameOfLife = new GameOfLife(this.props.length);
+        this.state = {
+            grid: this.gameOfLife.initialize()
+        };
+        this.changeGridState = this.changeGridState.bind(this);
```
Comment (Surzyn) [inline src/game-of-life/game.js]: remove this - change function changeGridState to arrow fun

File: src/game-of-life/game.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        this.clearGrid = this.clearGrid.bind(this);
```
Comment (Surzyn) [inline src/game-of-life/game.js]: remove this - change function clearGrid to arrow fun

**Review by Surzyn — APPROVED** [review]
good job :)

File: src/game-of-life/game.js (hunk 8 above)
Comment (dhudomiet) [reply src/game-of-life/game.js]: sorry this file shouldnt exists :)

File: src/game-of-life/game-of-life.js (hunk 7 above)
Comment (dhudomiet) [reply src/game-of-life/game-of-life.js]: done my Master

File: src/game-of-life/game-of-life.js (hunk 6 above)
Comment (dhudomiet) [reply src/game-of-life/game-of-life.js]: Done my Master

**Pull request merged**